pyungo/core.py

```diff
@@ -336,16 +344,29 @@ def calculate(self, data):
         t1 = dt.datetime.utcnow()
         LOGGER.info('Starting calculation...')
         self._data = Data(data)
-        self._data.check_inputs(self.sim_inputs, self.sim_outputs)
+        self._data.check_inputs(self.sim_inputs, self.sim_outputs, self.sim_kwargs)
         if not self._sorted_dep:
             self._topological_sort()
         for items in self._sorted_dep:
             # loading node with inputs
             for item in items:
                 node = self._get_node(item)
                 inputs = [i for i in node.inputs_without_constants]
+                kwarg_values = inspect.getargspec(node._fct).defaults
+
+                if node.kwargs and kwarg_values:
+                    kwarg_names = (inspect.getargspec(node._fct)
+                                   .args[-len(kwarg_values):])
+                    kwarg_defaults = {k: v for k, v
+                                      in zip(kwarg_names, kwarg_values)}
                 for inp in inputs:
-                    node.set_value_to_input(inp.name, self._data[inp.map])
+                    if (not inp.is_kwarg or
+                            (inp.is_kwarg and inp.map in self._data._inputs)):
+                        node.set_value_to_input(inp.name, self._data[inp.map])
+                    else:
+                        node.set_value_to_input(inp.name,
+                                                kwarg_defaults[inp.name])
```

Comment:
That makes sense! As a slight improvement, I would perhaps see a pre-loading of default value in the `Node` class when the function is registered. Then here, if we see a name that match a kwarg, we overwrite the loaded value.

Reply:
Great point! I'd be happy to improve this, maybe in another PR. :wink: